src/ocean/task/Scheduler.d

```diff
@@ -442,6 +442,13 @@ final class Scheduler
 
         This method must not be called outside of a task.
 
+        Because of how termination hooks are implemented, by the time `await`
+        returns, task object is not yet completely recycled - it will only
+        happen during next content switch. Caller of `await` must either ensure
+        task object live long enough for that or call
```

Comment:
`ensure that task object live long enough`? Not sure about this one, though. @gavin-norman-sociomantic ?